sys/posix/pthread/pthread_tls.c

```diff
@@ -16,10 +16,14 @@
  * @}
  */
 
-#include <malloc.h>
+
 
 #include "pthread.h"
 
+#ifndef __WITH_AVRLIBC__
```

Comment:
I meant you have `#include "pthread.h"`, line 21 and in `pthread.h` you already added:
```C
#ifndef __WITH_AVRLIBC__
#ifndef HAVE_MALLOC_H
#define HAVE_MALLOC_H 1
#endif
#endif
```
I'm also wondering if the `#ifndef HAVE_MALLOC_H` is superfluous.